mod-permissions/src/main/resources/domain/mod-permissions/features/permissions.feature

```diff
@@ -0,0 +1,398 @@
+Feature: Permissions tests
+
+  Background:
+    * url baseUrl
+    * call login testAdmin
+    * def commonHeaders =
+    """
+    {
+      "x-okapi-tenant": "#(testTenant)",
+      "x-okapi-token": "#(okapitoken)",
+      "Content-Type": "application/json",
+      "Accept": "application/json, text/plain"
+    }
+    """
+    * def optionsHeaders =
+    """
+    {
+      "Access-Control-Allow-Headers": "Content-Type, X-Okapi-Tenant, X-Okapi-Token, Authorization, X-Okapi-Request-Id, X-Okapi-Module-Id",
+      "Access-Control-Request-Method": "PUT, PATCH, DELETE, GET, POST",
+      "Access-Control-Allow-Origin": "*",
+      "Origin": "#(baseUrl)"
+    }
+    """
+    * configure lowerCaseResponseHeaders = true
+    * def testUserId = "00000000-1111-5555-9999-999999999998"
+    * def permissionsUserId = "00000000-1111-5555-9999-888888888885"
+
+  #
+  # Test permissions operations on users. This is similar to what happens in the browser when a new
+  # user is created or when a user's permissions are updated.
+  #  
+
+  # This emulates creating a user from scratch via the UI. We need a user without any permissions yet,
+  # so the existing test user (the one created in setup-users.feature) won't work because it already has
+  # permissions.
+  Scenario: Create a test user without any permissions yet
+    Given path 'users'
+    And headers commonHeaders
+    And request
+    """
+    {
+      "id": "#(testUserId)",
+      "username": "TestUser012",
+      "active": true
+    }
+    """
+    When method POST
+    Then status 201
+
+  Scenario: Create permissions for a user and check that the permission has been granted
+    * def perms = ["okapi.all"]
+ 
+    # Do a preflight request to emulate the browser.
+    Given path 'perms/users'
+    And headers karate.merge(commonHeaders, optionsHeaders)
+    When method OPTIONS
+    Then status 204
+    And match header access-control-allow-methods contains "POST"
```

Comment:
In general, do we need test OPTIONS in specific module, or should it be covered in Okapi tests?

Reply:
I think the OPTIONS requests are useful here. Including them uncovered interesting issues in our mod-login-saml tests for example. I think it's good to try to emulate what browser does as much as we can.

Reply:
Having OPTIONS integration tests for one module is fine, and for APIs where CORS is delegated from Okapi to the module, but not for all modules.